portal/views/teacher/teach_new.py

```diff
@@ -265,6 +268,107 @@ def teacher_delete_class_new(request, access_code):
 
 @login_required(login_url=reverse_lazy('login_new'))
 @user_passes_test(logged_in_as_teacher, login_url=reverse_lazy('login_new'))
+def teacher_edit_class_new(request, access_code):
+    klass = get_object_or_404(Class, access_code=access_code)
+
+    # check user authorised to see class
+    if request.user.new_teacher != klass.teacher:
+        raise Http404
+
+    external_requests_message = get_requests_message(klass)
+
+    if request.method == 'POST':
+        form = ClassEditForm(request.POST)
+        if form.is_valid():
+            return process_edit_class_form(request, klass, form)
+
+    else:
+        form = ClassEditForm(initial={
+            'name': klass.name,
+            'classmate_progress': klass.classmates_data_viewable,
+        })
+
+    return render(request, 'redesign/teach_new/teacher_edit_class_new.html',
+                  {'form': form,
+                   'class': klass,
+                   'external_requests_message': external_requests_message})
+
+
+def get_requests_message(klass):
+    if klass.always_accept_requests:
+        external_requests_message = 'This class is currently set to always accept requests.'
+    elif klass.accept_requests_until is not None and (klass.accept_requests_until - timezone.now()) >= timedelta():
```

Comment:
you might want to use a model method instead here? As I think this comparison is also used in play.py? https://docs.djangoproject.com/en/1.8/topics/db/models/#model-methods